src/components/WatchLater.jsx

```diff
@@ -0,0 +1,42 @@
+import { useSelector, useDispatch } from 'react-redux'
+import { Link } from 'react-router-dom'
+import watchLaterSlice from '../data/watchLaterSlice'
+import Movie from './Movie'
+import '../styles/starred.scss'
+
+const WatchLater = ({viewTrailer}) => {
+
+    const state = useSelector((state) => state)
+    const { watchLater } = state
+    const { remveAllWatchLater } = watchLaterSlice.actions
+    const dispatch = useDispatch()
+
+  return (
+    <div className="starred" data-testid="watch-later-div">
+      {watchLater.watchLaterMovies.length > 0 && (<div data-testid="watch-later-movies" className="starred-movies">
+        <h6 className="header">Watch Later List</h6>
+        <div className="row">
+        {watchLater.watchLaterMovies.map((movie) => (
```

Comment:
I would recommend using the MoviesGrid component instead of manually mapping the movie elements (<Movie />). This improves readability and makes the code easier to reuse.